src/main/java/ch/njol/skript/hooks/economy/expressions/ExprBalance.java

```diff
@@ -32,29 +32,22 @@
 import ch.njol.skript.hooks.VaultHook;
 import ch.njol.skript.hooks.economy.classes.Money;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Money")
 @Description("How much virtual money a player has (can be changed). This expression requires Vault and a compatible economy plugin to be installed.")
 @Examples({"message \"You have %player's money%\" # the currency name will be added automatically",
 		"remove 20$ from the player's balance # replace '$' by whatever currency you use",
-		"add 200 to the player's account # or omit the currency alltogether"})
+		"add 200 to the player's account # or omit the currency altogether"})
 @Since("2.0, 2.5 (offline player support)")
 @RequiredPlugins({"Vault", "a permission plugin that supports Vault"})
 public class ExprBalance extends SimplePropertyExpression<OfflinePlayer, Money> {
+
 	static {
 		register(ExprBalance.class, Money.class, "(money|balance|[bank] account)", "offlineplayers");
 	}
 	
-	@SuppressWarnings("deprecation")
 	@Override
-	public Money convert(final OfflinePlayer p) {
-		try {
-			return new Money(VaultHook.economy.getBalance(p));
-		}catch(Exception e){
-			return new Money(VaultHook.economy.getBalance(p.getName()));
-		}
```

Comment:
This seems like there was a scenario where calling getBalance(Player) would error while getBalance(String) wouldn't, can you find out why this fallback was added? For example, old vault versions

Reply:
I have reverted these changes just in case, I haven't yet got the time to go back to history and test why this was added but could be as you said something regarding old vault api or due to some old method of finding offline players failing.